File: kala-game-service/src/main/java/org/jesperancinha/games/kalagameservice/controller/KalaGameRegistrationController.java
package org.jesperancinha.games.kalagameservice.controller;

import org.jesperancinha.games.kalagameservice.service.PlayerService;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.PostMapping;
import org.springframework.web.bind.annotation.RestController;

import java.security.Principal;

@RestController
public class KalaGameRegistrationController {

    private final PlayerService playerService;

    public KalaGameRegistrationController(PlayerService playerService) {
        this.playerService = playerService;
    }

    @PostMapping("/")
    public void createPlayer(Principal principal) {
        playerService.createPlayer(principal.getName());
    }

    @GetMapping("/user")
    public String greetUser(Principal principal) {
        return String.format("Hello, %s", principal.getName());
    }
}

File: kala-game-service/src/main/kotlin/org/jesperancinha/games/kalagameservice/controller/KalaGameRegistrationController.kt
package org.jesperancinha.games.kalagameservice.controller

import org.springframework.web.bind.annotation.RestController
import org.springframework.web.bind.annotation.RequestMapping
import org.jesperancinha.games.kalagameservice.service.PlayerService
import org.springframework.web.bind.annotation.PostMapping
import java.security.Principal
import org.springframework.web.bind.annotation.GetMapping

@RestController
@RequestMapping("log")
class KalaGameRegistrationController(private val playerService: PlayerService) {
    @PostMapping("/")
    fun createPlayer(principal: Principal) {
        playerService.createPlayer(principal.name)
    }

    @GetMapping("user")
    fun greetUser(principal: Principal): String {
        return principal.name
    }
}